### PR TITLE
GAWB-2613 Upgrade Hail version

### PR DESCRIPTION
**Pull request by helgridly**
Tested. It lives!

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: jupyter-docker/Dockerfile [hunk 1]
```diff
@@ -79,10 +79,9 @@ RUN echo "deb $DEBIAN_REPO/debian jessie main\ndeb $DEBIAN_REPO/debian-security
  && wget -q https://storage.googleapis.com/hadoop-lib/gcs/gcs-connector-latest-hadoop2.jar -O $SPARK_HOME/gcs-connector.jar \
 
  # Hail
- && mkdir $HAIL_HOME && git clone https://github.com/broadinstitute/hail.git $HAIL_HOME \
- && cd $HAIL_HOME && git checkout $HAILHASH \
- && wget http://storage.googleapis.com/hail-common/$HAILJAR \
- && wget http://storage.googleapis.com/hail-common/$HAILZIP \
+ && mkdir $HAIL_HOME && cd $HAIL_HOME \
+ && wget http://storage.googleapis.com/hail-common/builds/0.1/jars/$HAILJAR \
+ && wget http://storage.googleapis.com/hail-common/distributions/0.1/$HAILZIP \
```
Comment (helgridly) [inline jupyter-docker/Dockerfile]: An attempt at optimization. I can't see where we use anything in the git repo. I also don't think we're using anything in the zip either (we're certainly never unzipping it). AFAICT It's just the jar that gets passed around.

Comment (helgridly) [reply jupyter-docker/Dockerfile]: The zip file is used in spark-defaults (which I found-and-replaced already, but didn't realise). Thanks Joel.

**Comment by jmthibault79** [comment]
Successfully built and pushed as *broadinstitute/leonardo:gawb-2613*

I have NOT deployed and tested.

**Comment by ansingh7115** [comment]
I built `cluster270-anu` with the above docker image and it seems to be working.
[https://leonardo.dsde-dev.broadinstitute.org/notebooks/broad-dsde-dev/custer270-anu/tree](
https://leonardo.dsde-dev.broadinstitute.org/notebooks/broad-dsde-dev/custer270-anu/tree)

Init stuff is here [https://console.cloud.google.com/storage/browser/custer270-anu-d84cc5cf-a2b5-473b-afbf-257d96ba8ca7](https://console.cloud.google.com/storage/browser/custer270-anu-d84cc5cf-a2b5-473b-afbf-257d96ba8ca7)

**Comment by jmthibault79** [comment]
Looks like we have a problem (Joel-Test on the above cluster):

screen shot 2017-09-25 at 10 26 23 am

**Comment by rtitle** [comment]
docker logs jupyter-server
```
17/09/25 14:24:01 ERROR spark.SparkContext: Error initializing SparkContext.
java.io.FileNotFoundException: File file:/hail/hail-0.1-$6f8598516c88-Spark-2.0.2.jar does not exist
	at org.apache.hadoop.fs.RawLocalFileSystem.getFileStatus(RawLocalFileSystem.java:520)
	at org.apache.hadoop.fs.FilterFileSystem.getFileStatus(FilterFileSystem.java:398)
	at org.apache.hadoop.fs.FileUtil.copy(FileUtil.java:337)
	at org.apache.hadoop.fs.FileUtil.copy(FileUtil.java:289)
	at org.apache.spark.deploy.yarn.Client.copyFileToRemote(Client.scala:340)
	at org.apache.spark.deploy.yarn.Client.org$apache$spark$deploy$yarn$Client$$distribute$1(Client.scala:433)
	at org.apache.spark.deploy.yarn.Client$$anonfun$prepareLocalResources$11$$anonfun$apply$6.apply(Client.scala:553)
	at org.apache.spark.deploy.yarn.Client$$anonfun$prepareLocalResources$11$$anonfun$apply$6.apply(Client.scala:552)
	at scala.collection.mutable.ArraySeq.foreach(ArraySeq.scala:74)
	at org.apache.spark.deploy.yarn.Client$$anonfun$prepareLocalResources$11.apply(Client.scala:552)
	at org.apache.spark.deploy.yarn.Client$$anonfun$prepareLocalResources$11.apply(Client.scala:551)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at org.apache.spark.deploy.yarn.Client.prepareLocalResources(Client.scala:551)
	at org.apache.spark.deploy.yarn.Client.createContainerLaunchContext(Client.scala:834)
	at org.apache.spark.deploy.yarn.Client.submitApplication(Client.scala:167)
	at org.apache.spark.scheduler.cluster.YarnClientSchedulerBackend.start(YarnClientSchedulerBackend.scala:56)
	at org.apache.spark.scheduler.TaskSchedulerImpl.start(TaskSchedulerImpl.scala:149)
	at org.apache.spark.SparkContext.<init>(SparkContext.scala:497)
	at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:240)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:236)
	at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
	at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
	at py4j.GatewayConnection.run(GatewayConnection.java:214)
	at java.lang.Thread.run(Thread.java:748)
17/09/25 14:24:01 WARN cluster.YarnSchedulerBackend$YarnSchedulerEndpoint: Attempted to request executors before the AM has registered!
17/09/25 14:24:01 WARN metrics.MetricsSystem: Stopping a MetricsSystem that is not running
```

File: jupyter-docker/spark-defaults.conf.template [hunk 2]
```diff
@@ -13,14 +13,14 @@ spark.submit.deployMode client
 
 # Distributes from master node to the working directories of executors
 
-spark.jars /spark/spark-2.0.2/gcs-connector.jar,/hail/hail-hail-is-master-all-spark2.0.2-[HAILHASH].jar
-spark.submit.pyFiles /hail/pyhail-hail-is-master-[HAILHASH].zip
+spark.jars /spark/spark-2.0.2/gcs-connector.jar,/hail/hail-0.1-$[HAILHASH]-Spark-2.0.2.jar
```
Comment (jmthibault79) [inline jupyter-docker/spark-defaults.conf.template]: looks like this is the problem.  Remove the `$` from all `[HAILHASH]`

**Comment by rtitle** [comment]
`/hail/hail-0.1-6f8598516c88-Spark-2.0.2.jar` _does_ exist (without the dollar sign). Prob need to remove the $ in `spark-defaults.conf.template`?

**Comment by jmthibault79** [comment]
Let's try again!  New docker image **broadinstitute/leonardo:gawb-2613-2**

**Comment by rtitle** [comment]
Aren't we using `broadinstitute/leonardo-notebooks` for the Jupyter stuff? I thought `broadinstitute/leonardo` was for the Leo app itself.

https://github.com/broadinstitute/firecloud-develop/blob/leonardo-configs/configs/leonardo/leonardo.conf.ctmpl#L10

**Comment by jmthibault79** [comment]
Good point.  It's now **broadinstitute/leonardo-notebooks:gawb-2613-2**

**Comment by ansingh7115** [comment]
Created cluster `cluster271-anu`
[https://leonardo.dsde-dev.broadinstitute.org/notebooks/broad-dsde-dev/cluster271-anu/tree?](https://leonardo.dsde-dev.broadinstitute.org/notebooks/broad-dsde-dev/cluster271-anu/tree?)
Init actions: [https://console.cloud.google.com/storage/browser/cluster271-anu-682468b2-02fb-4a2b-b1b0-f45ba3f1b42f?project=broad-dsde-dev&organizationId=548622027621](https://console.cloud.google.com/storage/browser/cluster271-anu-682468b2-02fb-4a2b-b1b0-f45ba3f1b42f?project=broad-dsde-dev&organizationId=548622027621)

**Pull request merged**